userspace/libscap/scap_procs.c

```diff
@@ -403,6 +403,72 @@ static int32_t scap_proc_fill_flimit(scap_t *handle, uint64_t tid, struct scap_t
 }
 #endif
 
+int32_t scap_proc_fill_overlayfs_root(scap_t *handle, struct scap_threadinfo *tinfo, const char *procdirname)
+{
+	FILE *mounts_file;
+	char mounts_path[SCAP_MAX_PATH_SIZE];
+	char line[SCAP_MAX_ARGS_SIZE];
+
+	tinfo->overlayfs_root[0] = '\0';
+
+	snprintf(mounts_path, sizeof(mounts_path), "%smounts", procdirname);
+	mounts_file = fopen(mounts_path, "r");
+	if(!mounts_file)
+	{
+		return SCAP_FAILURE;
+	}
+
+	if(!fgets(line, sizeof(line), mounts_file))
```

Comment:
We are only interested in first line of mounts file, because it is the line with the overlayfs in root, inside containers.